woocommerce/woocommerce
# Add experimental add product for task list

### All Submissions:






### Changes proposed in this Pull Request:



Closes #32634.

Implement experimental product page with two layouts:

Stack layout:

![Screen Shot 2022-04-27 at 16 56 57](https://user-images.githubusercontent.com/4344253/165481615-a6a92d45-5702-4175-bacc-3b3c55c4e34d.png)

Card layout:

![Screen Shot 2022-04-27 at 16 57 22](https://user-images.githubusercontent.com/4344253/165481629-7f831851-a25f-41dc-a70d-45f6751df924.png)



### How to test the changes in this Pull Request:

1. Use a fresh site
2. Open the `plugins/woocommerce/client/admin/config/development.json` and change `experimental-products-task` to true
3. Run `pnpm nx dev woocommerce-admin`
4. Skip OBW
5. Go to Woocommerce > Home
6. Click **"Add my products"** task
7. Observe that "Physical product, Variable product, Grouped product" are shown by default. The other ones are collapsed.
8. Go to OBW
9. Select "Physical products" in the product types step
10. Go to Woocommerce > Home
11. Click **"Add my products"** task
12. Observe that "Physical product, Variable product, Grouped product" are shown.
13. Click "View more product types" button.
14. Should display all product type
15. Repeat 9~12 with other configurations below

The visible product types depend on what the user declared during the signup (step 3 - Product types - What type of products will be listed?). The other ones are collapsed. 

> Selected (signup): Physical products
> Surfaced (add product page): Physical product, Variable product, Grouped product

> Selected (signup): Physical products, Subscriptions
> Surfaced (add product page): Physical product, Subscription product

> Selected (signup): Physical products, Downloads
> Surfaced (add product page): Physical product, Digital product

> Selected (signup): Physical products, Downloads, Subscriptions
> Surfaced (add product page): Physical product, Digital product, Subscription product

> Selected: Downloads
> Surfaced: Digital product

> Selected: Downloads, Subscriptions
> Surfaced: Digital product, Subscription product

> Selected: Subscriptions
> Surfaced: Subscription product



**Card Layout**

1. Change `SHOW_STACK_LAYOUT` to false in [this line](https://github.com/woocommerce/woocommerce/blob/8892b61f3d2db064d69e24913a48514319cc2582/plugins/woocommerce-admin/client/tasks/fills/experimental-products/index.tsx#L24).
2.  Run `pnpm nx dev woocommerce-admin`
3. Go to **"Add my products"** task
4. Observe that page shows the product types in card layout.

### Other information:




### FOR PR REVIEWER ONLY:
